### PR TITLE
Rewrite symbolic_locals for torch.return_types

### PR DESCRIPTION
**Pull request by anijain2305**
As Title

### CODE REVIEW

File: tests/test_repros.py [hunk 1]
```diff
@@ -1369,3 +1369,22 @@ def forward(self, x):
...
+    def test_sort_out(self):
+
+        dtype = torch.float32
+        device = "cpu"
+
+        def fn():
+            tensor = torch.randn((3, 5), dtype=dtype, device=device)[:, 0]
+            values = torch.tensor(0, dtype=dtype, device=device)
+            indices = torch.tensor(0, dtype=torch.long, device=device)
+            torch.sort(tensor, out=(values, indices))
```
[the remark this answers is not in the document]
Comment (anijain2305) [reply tests/test_repros.py:1382]: Hmm, changing the test  to 
~~~
        def fn():
            tensor = torch.randn((3, 5), dtype=dtype, device=device)[:, 0]
            values1 = torch.tensor(0, dtype=dtype, device=device)
            indices1 = torch.tensor(0, dtype=torch.long, device=device)
            torch.sort(tensor, out=(values1, indices1))
            self.assertEqual(values1.stride(), (1,))
            self.assertEqual(indices1.stride(), (1,))
~~~

fails both w and w/o my changes. Basically, this is the problem with out_variants, and we need to somehow recognize the outputs and rewrite the locals.

[the remark this answers is not in the document]
Comment (jansel) [reply tests/test_repros.py:1382]: Hrm, `torch.sort` should have no way of knowing what your local variable names are...

```
>>> tensor = torch.randn((3, 5))[:, 0]
>>> values1 = torch.tensor(0.0)
>>> indices1 = torch.tensor(0)
>>> print(id(values1), id(indices1))
139939847176896 139939847177056
>>> torch.sort(tensor, out=(values1, indices1))
torch.return_types.sort_out(
values=tensor([-2.1302,  0.8134,  1.5392]),
indices=tensor([1, 2, 0]))
>>> print(id(values1), id(indices1))
139939847176896 139939847177056
>>> print(values1)
tensor([-2.1302,  0.8134,  1.5392])
>>> print(indices1)
tensor([1, 2, 0])
>>> 
```

[the remark this answers is not in the document]
Comment (anijain2305) [reply tests/test_repros.py:1382]: I think we need special case handling for any torch op with an `out=` argument. I tried a couple more examples (torch.special.expit, and all mutate the ops in-place.

[the remark this answers is not in the document]
Comment (anijain2305) [reply tests/test_repros.py]: Refactored to special case on the `out` variant. What do you think?

File: torchdynamo/symbolic_convert.py [hunk 2]
```diff
@@ -1127,6 +1127,12 @@ def frame_summary(self):
             lookup_line=False,
         )
 
+    def find_symbolic_locals_name(self, tensor_variable):
+        for key, value in self.symbolic_locals.items():
+            if value == tensor_variable:
```
[the remark this answers is not in the document]
Comment (anijain2305) [reply torchdynamo/symbolic_convert.py:1132]: Hmm. Good question. And something I should have known before

The default __eq__ (if you don't supply anything) is `is` operator. And `is` checks if the two objects are identical (perhaps it uses `id` to check if they are identical). So, in this case, `==` equals `is`, which is accidentally what we want as well. I can replace `==` with `is` to be more precise. Is that ok?

**Pull request merged**